### PR TITLE
fix(workspace): parse actual XML output from mvn help:evaluate

### PR DESCRIPTION
**Pull request by Strum355**
## Summary
- `mvn help:evaluate -Dexpression=project.modules` returns XML (`<strings><string>module-a</string>...</strings>`), not the bracket format (`[module-a, module-b]`) the original parser assumed
- Replaced regex-based bracket parser with XML DOM parser
- Cleaned up inline fully-qualified types into proper imports
- Replaced mocked integration tests with real Maven CLI integration tests

## Test plan
- [x] Verified actual `mvn help:evaluate` output against the multi-module test fixture
- [x] Unit tests pass (`MavenWorkspaceDiscoveryTest`)
- [x] Integration tests pass (`MavenWorkspaceDiscoveryIT`)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by Sourcery

Update Maven workspace discovery to handle actual XML output from Maven and validate it via real CLI integration tests.

New Features:
- Add Maven CLI-based integration tests for workspace manifest discovery across different multi-module layouts and ignore patterns.

Bug Fixes:
- Fix Maven module list parsing to correctly handle the XML format returned by `mvn help:evaluate -Dexpression=project.modules`, including empty and malformed inputs.

Enhancements:
- Simplify unit tests for Maven module list parsing to focus on XML-based scenarios and edge cases.
- Adjust behavior when Maven cannot resolve declared modules so that discovery degrades gracefully to the root pom where appropriate.

Tests:
- Replace mocked Maven workspace discovery tests with integration tests that run against real Maven projects using the Maven executable on PATH.

### CODE REVIEW

File: src/main/java/io/github/guacsec/trustifyda/impl/ExhortApi.java [hunk 1]
```diff
@@ -1198,27 +1201,37 @@ private List<String> listMavenModules(Path dir, String mvnBin) {
...
-    if (raw == null || raw.isEmpty()) {
+    if (raw == null || raw.isEmpty() || "null".equals(raw.trim())) {
       return Collections.emptyList();
     }
-    // Expected format: [module-a, module-b, ...]
-    java.util.regex.Matcher matcher =
-        java.util.regex.Pattern.compile("^\\[(.+)]$").matcher(raw.trim());
-    if (!matcher.matches()) {
+    String trimmed = raw.trim();
+    if (!trimmed.startsWith("<") || trimmed.startsWith("<modules")) {
```
Comment (ruromero) [inline src/main/java/io/github/guacsec/trustifyda/impl/ExhortApi.java]: why not directly set
```suggestion
    if (!trimmed.startsWith("<strings")) {
```

**Review by ruromero — CHANGES_REQUESTED** [review]
I only added one comment but "thumbed-up" some sourcery-ai important feedback.

**Review by a-oren — APPROVED** [review]

**Review by ruromero — APPROVED** [review]
